ChatSX/components/Poll/index.tsx

```diff
@@ -0,0 +1,1689 @@
+          <Pressable onPress={() => {}} style={[styles.modalView]}>
+            <View style={styles.alignModalElements}>
+              <CreatePollScreen
+                handleModalClose={handleModalClose}
+                chatroomID={chatroomID}
+              />
+            </View>
+          </Pressable>
+        </View>
+      </Pressable>
+    </Modal>
+  );
+};
+
+export default CreatePollModal;
+
+const CreatePollScreen = ({navigation, route}: any) => {
+  const [question, setQuestion] = useState<string>('');
+  const [optionsArray, setOptionsArray] = useState<any>([]);
+  const [showAdvancedOption, setShowAdvancedOption] = useState<boolean>(false);
+  const [addOptionsEnabled, setAddOptionsEnabled] = useState<boolean>(false);
+  const [anonymousPollEnabled, setAnonymousPollEnabled] =
+    useState<boolean>(false);
+  const [liveResultsEnabled, setLiveResultsEnabled] = useState<boolean>(false);
+  const [userVoteFor, setUserVoteFor] = useState<number>(1);
+  const [voteAllowedPerUser, setVoteAllowedPerUser] = useState<number>(1);
+  const [date, setDate] = useState('');
+  const [mode, setMode] = useState<string>('');
+  const [show, setShow] = useState<boolean>(false);
+  const [time, setTime] = useState(new Date());
+  // const [isUserVoteForModal, setIsUserVoteForModal] = useState(false)
+  const [isActionAlertModalVisible, setIsActionAlertModalVisible] =
+    useState(false);
+  const [isOptionAlertModalVisible, setIsOptionAlertModalVisible] =
+    useState(false);
+  const [userVoteForOptionsArrValue, setUserVoteForOptionsArrValue] = useState(
+    [],
+  );
+
+  const userCanVoteForArr = ['Exactly', 'At max', 'At least'];
+
+  const dispatch = useAppDispatch();
+  const {chatroomID} = route.params;
+
+  const setInitialHeader = () => {
+    navigation.setOptions({
+      title: 'New Poll',
+      headerShadowVisible: false,
+      headerTitleStyle: [styles.font, styles.newPollText],
+      headerStyle: {backgroundColor: styles.lightGreyThumb.color},
+      headerLeft: () => (
+        <View style={[styles.alignRow, styles.header]}>
+          <TouchableOpacity
+            onPress={() => {
+              navigation.goBack();
+            }}
+            style={styles.viewStyle}>
+            <Text style={[styles.font]}>Cancel</Text>
+          </TouchableOpacity>
+        </View>
+      ),
+    });
+  };
+
+  // to set header of the component
+  useEffect(() => {
+    setInitialHeader();
+  }, []);
+
+  // to set initial poll options
+  useEffect(() => {
+    let id_1 = uuid.v4();
+    let id_2 = uuid.v4();
+    let initialOptionArray = [
+      {
+        id: id_1,
+        text: '',
+      },
+      {
+        id: id_2,
+        text: '',
+      },
+    ];
+    setOptionsArray(initialOptionArray);
+  }, []);
+
+  // this function formats the date in "DD/MM/YYYY hh:mm" format
+  const formatDate = (date: any, time: any) => {
+    let formattedTime = moment(date).format('DD/MM/YYYY hh:mm');
+    return formattedTime;
+  };
```

Comment:
Also, why this functions in UI file, it should be in some `DateTime` util class, move this there?